crates/agent/src/discovers.rs

```diff
@@ -335,11 +338,28 @@ impl DiscoverHandler {
 
         // Deeply merge the capture and its bindings.
         let capture_name = models::Capture::new(capture_name);
+        let existing_capture = catalog.captures.remove(&capture_name);
+        // This is a hack to prevent autoDiscovers from overwriting changes to the capture endpoint.
+        // Background discovers get prioritized behind interactive publications, so it's possible
+        // for an interactive publication to update the endpoint config after the `discovers` row
+        // has already been created but before it's been processed. This detects that condition and
+        // bails, so we can try again on the next auto-discover.
+        if existing_capture
+            .as_ref()
+            .is_some_and(|spec| background && is_endpoint_changed(&spec.endpoint, &endpoint))
```

Comment:
Why wouldn't we want this to apply to all publications? That is: if the proverbial Alice starts a publication that changes the endpoint config, and Bob starts another one that doesn't before Alice's publication finishes, won't Alice's change get clobbered? Feels like bob's publication should fail with a retryable error in this case (and maybe even auto-retry in the future?)

Reply:
If we did it for all publications, it would break when you edit a capture's endpoint config and then hit the "refresh bindings" button. The alice/bob scenario is legit a problem (though it's less of a problem with interactive publications because they generally don't sit in queued status anywhere near as long). But it's not new, and not exacerbated by this hack. Ultimately, we need discovers to set the `expect_pub_id` to protect against that sort of scenario. It doesn't today, partly because a `models::Catalog` historically had no way of representing it.

Reply:
>If we did it for all publications, it would break when you edit a capture's endpoint config and then hit the "refresh bindings" button.

Oh... yeah good point.

>But it's not new, and not exacerbated by this hack.
Yep, sounds good 👍